src/aind_codeocean_utils/codeocean_job.py

```diff
@@ -0,0 +1,582 @@
+"""Module with generic Code Ocean job"""
+import logging
+import time
+from datetime import datetime
+from typing import Dict, List, Optional, Tuple, Union
+
+import requests
+from aind_codeocean_api.codeocean import CodeOceanClient
+from aind_codeocean_api.models.computations_requests import RunCapsuleRequest
+from aind_codeocean_api.models.data_assets_requests import (
+    CreateDataAssetRequest,
+    Source,
+    Sources,
+)
+
+LOG_FMT = "%(asctime)s %(message)s"
+LOG_DATE_FMT = "%Y-%m-%d %H:%M"
+
+logging.basicConfig(format=LOG_FMT, datefmt=LOG_DATE_FMT)
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class CodeOceanJob:
+    """
+    This class contains convenient methods to register data assets,
+    run capsules, and capture results.
+    """
+
+    default_datetime_format = "%Y-%m-%d_%H-%M-%S"
```

Comment:
We might want to pull in the regex pattern from aind-data-schema

Reply:
where exactly?

Reply:
In theory, it's supposed to maintained here, but it's buried in one of the methods I think: https://github.com/AllenNeuralDynamics/aind-data-schema/blob/main/src/aind_data_schema/core/data_description.py